packages/eslint-plugin/src/rules/switch-exhaustiveness-check.ts

```diff
@@ -69,14 +69,23 @@ export default createRule({
         let caseTest = checker.typeToString(missingBranchType);
 
         if (
-          symbolName &&
           (missingBranchName || missingBranchName === '') &&
           requiresQuoting(missingBranchName.toString(), compilerOptions.target)
         ) {
-          caseTest = `${symbolName}['${missingBranchName}']`;
+          const requiresBackticks = missingBranchName?.match(/[\r\n]/g);
+
+          caseTest = requiresBackticks
+            ? `${symbolName}[\`${missingBranchName}\`]`
+            : `${symbolName}['${missingBranchName}']`;
```

Comment:
I don't know if it's discussed yet, but if we are escaping characters in the error message anyway, I would prefer text here to also be escaped instead of wrapping in backticks. Otherwise, you would also need to deal with `` ` `` in the text. (Do we even deal with `'` correctly?)

Reply:
OK, so your suggested case breaks the code (nice catch!)
But I'm not sure that I understand what you mean in this comment, can you maybe show some pseudo code and/or cases?

Reply:
It means we simply output `case '${escapedCaseTest}': { throw new Error("Not implemented yet: '${escapedCaseTest}' case") }`. One consistent string literal to be used in both places.

In terms of cases, it means we output:

```js
switch (a) {
  case 'a\nb': { throw new Error("Not implemented yet: 'a\nb' case") }
}
```

Instead of:

```js
switch (a) {
  case `a
b`: { throw new Error("Not implemented yet: 'a\nb' case") }
}
```

I'm sure no one writes code like the latter, especially with deeper indentations.

Reply:
I'm not sure that we can use the same `escapedCaseTest` variable, because we still need to re-escape It before using it in the error, right?

Reply:
Have a look at [this one](https://github.com/typescript-eslint/typescript-eslint/pull/7806/commits/c63f866d29df0f0bd5998674ec943589108c1bfa)

Is it what you mean?